### PR TITLE
feat: flexible engines mounting

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #[issue_no]

## Summary of Changes

This PR adds a physical model for engine mounts, so that engines will wobble when plane is experiencing sudden accelerations.

## References

https://youtu.be/O-qnWaqDwxg?t=257

## Additional context

Discord username (if different from GitHub):

## Testing instructions

No tests expected for now, this will be tested ingame on A380 plane

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/systems/systems/src/engine/engine_wing_flex.rs [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        let mut all_engines: Vec<EngineFlexPhysics> = vec![];
+        for engine_number in 1..=N {
+            all_engines.push(EngineFlexPhysics::new(context, engine_number));
+        }
+
+        Self {
+            engines_flex_updater: MaxStepLoop::new(Self::ENGINES_FLEX_SIM_TIME_STEP),
+            engines_flex: all_engines,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (crocket63) [reply src/systems/systems/src/engine/engine_wing_flex.rs:157]: I tried [EngineFlexPhysics; N] a lot, couldn't find any way so it compiles.... :(

File: src/systems/systems/src/engine/engine_wing_flex.rs [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+            self.anisotropic_damping_constant[0] = reader.read(&self.lateral_damping_id);
+            self.anisotropic_damping_constant[1] = reader.read(&self.vertical_damping_id);
+
+            // Z dimension not really used we use same param as Y damping
+            self.anisotropic_damping_constant[2] = reader.read(&self.vertical_damping_id);
+        }
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(&self.x_position_id, self.animation_position());
```
[the remark this answers is not in the document]
Comment (crocket63) [reply src/systems/systems/src/engine/engine_wing_flex.rs:125]: Don't think there's any impact there as the calculation would have to be done once anyway at one point, be it at write or at update time.

**Comment by Gurgel100** [comment]
Just out of interest, are you considering to consider engine vibration later as well?

**Review by Gurgel100 — APPROVED** [review]
LGTM

**Comment by crocket63** [comment]
> Just out of interest, are you considering to consider engine vibration later as well?

Not sure it would be relevant, would probably be higher frequencies with less amplitude, so more energy but not visible as the structural wobble. I can be wrong though :p

**Comment by Gurgel100** [comment]
> Not sure it would be relevant, would probably be higher frequencies with less amplitude, so more energy but not visible as the structural wobble. I can be wrong though :p

Well at least in case of a fatal engine failure the vibration certainly make the engine wobble :D

**Comment by crocket63** [comment]
Oh yeah when it explodes yes lol. I bet we'll have to figure out something function of speed then to excite the animation :)

**Pull request merged**